### PR TITLE
Implement '/meta'

### PR DESCRIPTION
**Pull request by pcasaretto**
Hey guys,

Learning go and currently using go-octokit as a dependency for another project.
I wanted to contribute back and also scratch my own itch :) .
Opening this PR with WIP to get early feedback.
The code currently works.

Some specific questions on the code itself.

Thanks!

### CODE REVIEW

File: octokit/meta.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// A service to return information about GitHub.com, the service.
+type MetaService struct {
+	client *Client
+	URL    *url.URL
+}
+
+// Get the user search results based on MetaService#URL
+//
+// https://developer.github.com/v3/meta/#meta
+func (m *MetaService) One() (meta Meta, result *Result) {
```
Comment (pcasaretto) [inline octokit/meta.go]: is `One` a good name for this function?

File: octokit/meta.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	result = m.client.get(m.URL, &meta)
+	return
+}
+
+type Meta struct {
+	*hypermedia.HALResource
+
+	VerifiablePasswordAuthentication bool     `json:"verifiable_password_authentication,omitempty"`
+	GithubServicesSha                string   `json:"github_services_sha,omitempty"`
+	Hooks                            []string `json:"hooks,omitempty"`
```
Comment (pcasaretto) [inline octokit/meta.go]: should we deliver arrays of `*net.IPNet`s or `string`s?

File: octokit/meta.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	Git                              []string `json:"git,omitempty"`
+	Pages                            []string `json:"pages,omitempty"`
```
Comment (pcasaretto) [inline octokit/meta.go]: The current API seems to have an extra `importer` field. Is the documentation out of date?
Should we include it here?

File: octokit/meta.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+// Get the user search results based on MetaService#URL
+//
+// https://developer.github.com/v3/meta/#meta
+func (m *MetaService) One() (meta Meta, result *Result) {
+	result = m.client.get(m.URL, &meta)
+	return
+}
+
+type Meta struct {
```
Comment (pcasaretto) [inline octokit/meta.go]: Is this a good name for this struct?

**Comment by brodock** [comment]
@pengwynn can you please provide some feedback here? We are implementing other missing pieces and would like to know if you are interested in more PRs.

File: octokit/meta.go (hunk 1 above)
Comment (pengwynn) [inline octokit/meta.go]: Thanks for checking. `One` isn't a great fit here, is it? Wondering if we should forgo the service and just have `client.Meta` return an `ApiInfo` struct? Thoughts?

File: octokit/meta.go (hunk 4 above)
Comment (pengwynn) [inline octokit/meta.go]: I think it depends on what we do for the method name. https://github.com/octokit/go-octokit/pull/118/files#r36671782

File: octokit/meta.go (hunk 3 above)
Comment (pengwynn) [inline octokit/meta.go]: We should include it and update the docs. /cc @spraints @jonmagic

File: octokit/meta.go (hunk 2 above)
Comment (pengwynn) [inline octokit/meta.go]: Oooh I like the `IPNet` idea. :+1:

**Comment by pengwynn** [comment]
@pcasaretto @brodock: My apologies, I was on vacation when this landed. Was buried in my :star: folder.

I left some feedback. Thanks for the patch!

**Comment by pcasaretto** [comment]
Thanks for the feedback @pengwynn. Tried to incorporate all of it in the most recent commit.
Apart from the obvious error cases, should I focus on any specific test cases?

File: octokit/meta.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/jingweno/go-sawyer/hypermedia"
+)
+
+// https://developer.github.com/v3/meta/
+var (
+	MetaURL = Hyperlink("/meta")
+)
+
+// Meta return an APIInfo with the current API meta information
+func (c *Client) Meta(uri *Hyperlink) (info APIInfo, result *Result) {
```
Comment (pcasaretto) [inline octokit/meta.go]: I was a little confused if I should accept a `*url.URL`, a `*Hyperlink`, or nothing at all here.

**Comment by pcasaretto** [comment]
I've added tests for the basic cases, but there is lack of coverage where the IPs and IPNets get parsed.
Do you think it's important to test that?
I did try but I could't stub the request on the same test and had to add two different json fixtures. I figured it wasn't worth the trouble.

File: octokit/meta.go (hunk 5 above)
Comment (pengwynn) [inline octokit/meta.go]: I think this looks good.

**Comment by pengwynn** [comment]
> I've added tests for the basic cases, but there is lack of coverage where the IPs and IPNets get parsed.
Do you think it's important to test that?

I think this is fine for now. Thanks for the patch. :cake:

**Pull request merged**